arcgis-ios-sdk-samples/Scenes/Animate images with image overlay/AnimateImagesWithImageOverlayViewController.swift

```diff
@@ -0,0 +1,215 @@
+// Copyright 2020 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class AnimateImagesWithImageOverlayViewController: UIViewController {
+    // MARK: Storyboard views
+    
+    /// The button to play image overlay animation.
+    @IBOutlet var playButtonItem: UIBarButtonItem!
+    /// The button to pause image overlay animation.
+    @IBOutlet var pauseButtonItem: UIBarButtonItem!
+    /// The button to choose a playback speed for the animation.
+    @IBOutlet weak var speedButtonItem: UIBarButtonItem!
+    /// The toolbar in the view controller.
+    @IBOutlet weak var toolbar: UIToolbar!
+    /// The label to display opacity level.
+    @IBOutlet weak var opacityLabel: UILabel!
+    
+    /// The slider to change opacity level, from transparent 0% to opaque 100%.
+    @IBOutlet weak var opacitySlider: UISlider! {
+        didSet {
+            sliderValueChanged(opacitySlider)
+        }
+    }
+    
+    /// The scene view managed by the view controller.
+    @IBOutlet weak var sceneView: AGSSceneView! {
+        didSet {
+            sceneView.scene = makeScene()
+            let point = AGSPoint(x: -116.621, y: 24.7773, z: 856977.0, spatialReference: .wgs84())
+            let camera = AGSCamera(location: point, heading: 353.994, pitch: 48.5495, roll: 0)
+            sceneView.setViewpointCamera(camera)
+            sceneView.imageOverlays.add(imageOverlay)
+        }
+    }
+    
+    // MARK: Instance properties
+    
+    /// The frame rate in the unit of frame per second.
+    var frameRate = 60
+    /// The image overlay to show image frames.
+    let imageOverlay = AGSImageOverlay()
+    /// A timer to synchronize image frame animation to the refresh rate of the display.
+    var displaylink: CADisplayLink!
+    
+    /// An iterator to hold and loop through the overlay images.
+    private lazy var imagesIterator: CircularIterator<UIImage> = {
+        // Get the URLs to images added to the project's folder reference.
+        var imageURLs = Bundle.main.urls(forResourcesWithExtension: "png", subdirectory: "PacificSouthWest") ?? []
+        imageURLs.sort { $0.lastPathComponent < $1.lastPathComponent }
+        return CircularIterator(
+            elements: imageURLs.map { UIImage(contentsOfFile: $0.path)! }
+        )
```

Comment:
This isn't a big deal right here, but I've found it beneficial to use immutable values whenever possible. `imageURLs` doesn't need to be mutable.

```suggestion
        let imageURLs = Bundle.main.urls(forResourcesWithExtension: "png", subdirectory: "PacificSouthWest") ?? []
        let images = imageURLs
            .sorted { $0.lastPathComponent < $1.lastPathComponent }
            .map { UIImage(contentsOfFile: $0.path)! }
        return CircularIterator(elements: images)
```